database/database.go

```diff
@@ -0,0 +1,88 @@
+package database
+
+import (
+	"dashboard/model"
+	"database/sql"
+	"log"
+
+	"golang.org/x/crypto/bcrypt"
+)
+
+var db *sql.DB
+
+//InitialiseDb to start db server
+func InitialiseDb() {
+	var err error
+	db, err = sql.Open("mysql", "root:mysql1234@tcp(localhost:3306)/dashboard")
+	if err != nil {
+		log.Panic(err.Error())
+	}
+}
+
+//CloseDb to close DB connection
+func CloseDb() {
+	db.Close()
+}
+
+//GetUserByUsername func for sign_up
+func GetUserByUsername(user *model.UserProfile) (*model.UserProfile, error) {
+	var err error
+	stmtOut, err := db.Prepare("SELECT username FROM user_profile WHERE username  = ? ")
+	if err != nil {
+		panic(err.Error())

```

Comment:
Dont' use panic or log.Panic it crashes the server. Your server will be down after one invalid request.
Instead do **return nil, err**

Reply:
only in this function or any ?